src/WebJobs.Script.WebHost/Extensions/FunctionMetadataExtensions.cs

```diff
@@ -0,0 +1,212 @@
+            Path.Combine(config.TestDataPath, $"{functionName}.dat");
+
+        private static async Task<JObject> GetFunctionConfig(string path)
+        {
+            try
+            {
+                if (FileUtility.FileExists(path))
+                {
+                    return JObject.Parse(await FileUtility.ReadAsync(path));
+                }
+            }
+            catch
+            {
+                // no-op
+            }
+
+            // If there are any errors parsing function.json return an empty object.
+            // This is current kudu behavior.
+            // TODO: add an error field that can be used to communicate the JSON parse error.
+            return new JObject();
+        }
+
+        private static async Task<string> GetTestData(string testDataPath, ScriptHostConfiguration config)
+        {
+            if (!File.Exists(testDataPath))
+            {
+                FileUtility.EnsureDirectoryExists(Path.GetDirectoryName(testDataPath));
+                await FileUtility.WriteAsync(testDataPath, string.Empty);
+            }
+
+            return await FileUtility.ReadAsync(testDataPath);
+        }
+
+        internal static Uri GetFunctionInvokeUrlTemplate(string baseUrl, FunctionMetadata functionMetadata, string routePrefix)
+        {
+            var httpBinding = functionMetadata.InputBindings.FirstOrDefault(p => string.Compare(p.Type, "httpTrigger", StringComparison.OrdinalIgnoreCase) == 0);
+
+            if (httpBinding != null)
+            {
+                string customRoute = null;
+                if (httpBinding.Raw != null && httpBinding.Raw.TryGetValue("route", StringComparison.OrdinalIgnoreCase, out JToken value))
+                {
+                    // a custom route is specified
+                    customRoute = (string)value;
+                }
+
+                string uriString = baseUrl.TrimEnd('/');
+                if (!string.IsNullOrEmpty(routePrefix))
+                {
+                    uriString += $"/{routePrefix.TrimEnd('/')}";
+                }
+
+                if (!string.IsNullOrEmpty(customRoute))
+                {
+                    uriString += $"/{customRoute}";
+                }
+                else
+                {
+                    uriString += $"/{functionMetadata.Name}";
+                }
+
+                return new Uri(uriString.ToLowerInvariant());
+            }
+
+            return null;
+        }
+
+        internal static string GetScmBaseUrl()
+        {
+            // VFS APIs aren't exposed by the v1 runtime, so we must redirect
+            // to the SCM site
+            string siteName = Environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteName);
+            string scmHostName;
+            if (!string.IsNullOrEmpty(siteName))
+            {
+                scmHostName = siteName + ".scm.azurewebsites.net";
```

Comment:
Is this a safe way to form this SCM url?

Reply:
Sorry I just saw this. I don't know of another way unfortunately. @suwatch do you know if there is a way to build the scm url that would work both in a private stamp and in prod?

Reply:
I also considered starting with WEBSITE_HOSTNAME but I'm leary of attempting to insert ".scm" into the host in the correct place.